aten/src/ATen/core/ivalue_inl.h

```diff
@@ -139,7 +139,9 @@ inline c10::complex<double> IValue::toComplexDouble() const {
   return (*ptr).val;
 }
 inline at::Tensor IValue::toTensor() && {
-  AT_ASSERT(isTensor(), "Expected Tensor but got ", tagKind());
+  if (C10_UNLIKELY(!isTensor())) {
```

Comment:
Out of curiosity, `isTensor()` should already returns a bool, what's the reason of using `C10_UNLIKELY` here?

Reply:
It's a branch prediction hint. Causes the compiler to generate code that optimizes for the condition being false.